[LLVM][AMDGPU] MC support for ds_bpermute_fi_b32

Acim-Maravic opened the pull request:
Added assembler/disassembler support for ds_bpermute_fi_b32 instruction, as well as tests.

arsenm commented on llvm/lib/Target/AMDGPU/GCNHazardRecognizer.cpp:
```diff
@@ -150,6 +150,7 @@ static bool isSendMsgTraceDataOrGDS(const SIInstrInfo &TII,
   case AMDGPU::DS_NOP:
   case AMDGPU::DS_PERMUTE_B32:
   case AMDGPU::DS_BPERMUTE_B32:
+  case AMDGPU::DS_BPERMUTE_FI_B32:
```
This is more than MC and untested

Acim-Maravic replied:
Removed.

jayfoad reviewed (approved):
LGTM with the hazard recognizer part removed.

Acim-Maravic committed:
[LLVM][AMDGPU] MC support for ds_bpermute_fi_b32

Added assembler/disassembler support for ds_bpermute_fi_b32 instruction,
as well as tests.

Acim-Maravic commented:
> LGTM with the hazard recognizer part removed.

Hazard recognizer part is removed.
